kernel/kernel-api/src/main/java/io/delta/kernel/internal/files/ParsedCheckpointData.java

```diff
@@ -16,97 +16,104 @@
 
 package io.delta.kernel.internal.files;
 
-import static io.delta.kernel.internal.util.Preconditions.checkArgument;
-
 import io.delta.kernel.data.ColumnarBatch;
-import io.delta.kernel.internal.util.FileNames;
 import io.delta.kernel.utils.FileStatus;
 import java.util.Optional;
 
-public class ParsedCheckpointData extends ParsedLogData
+/**
+ * Abstract checkpoint file that contains a complete snapshot of table state at a specific version.
+ */
+public abstract class ParsedCheckpointData extends ParsedLogData
     implements Comparable<ParsedCheckpointData> {
 
-  public static ParsedCheckpointData forFileStatus(FileStatus fileStatus) {
-    final String path = fileStatus.getPath();
-
-    if (FileNames.isMultiPartCheckpointFile(path)) {
-      return ParsedMultiPartCheckpointData.forFileStatus(fileStatus);
-    }
-
-    final long version;
-    final ParsedLogType type;
-
-    if (FileNames.isClassicCheckpointFile(path)) {
-      version = FileNames.checkpointVersion(path);
-      type = ParsedLogType.CLASSIC_CHECKPOINT;
-    } else if (FileNames.isV2CheckpointFile(path)) {
-      version = FileNames.checkpointVersion(path);
-      type = ParsedLogType.V2_CHECKPOINT;
-    } else {
-      throw new IllegalArgumentException("File is not a recognized checkpoint type: " + path);
-    }
-
-    return new ParsedCheckpointData(version, type, Optional.of(fileStatus), Optional.empty());
+  protected ParsedCheckpointData(
+      long version, Optional<FileStatus> fileStatusOpt, Optional<ColumnarBatch> inlineDataOpt) {
+    super(version, fileStatusOpt, inlineDataOpt);
   }
 
-  public static ParsedCheckpointData forInlineData(
-      long version, ParsedLogType type, ColumnarBatch inlineData) {
-    if (type == ParsedLogType.MULTIPART_CHECKPOINT) {
-      throw new IllegalArgumentException(
-          "For MULTIPART_CHECKPOINT, use ParsedMultiPartCheckpointData.forInlineData() instead");
-    }
+  /**
+   * Returns the priority of this checkpoint type for Kernel's checkpoint preference ranking. Higher
+   * values indicate higher priority (preferred checkpoints).
+   *
+   * <p>This priority is only used to break ties when multiple checkpoint types exist at the same
+   * version. The ordering is based on safety and parallelizability of reading:
+   *
+   * <ul>
+   *   <li>V2 (priority 2): Best and safest checkpoint format
```

Comment:
Wondering if we could define a enum that are comparable or just add ranking ability to ParseLogData enum? If we add 4th type of checkpoint in the future, we need to update the code/comments everywhere.

public enum CheckpointType implements Comparable<CheckpointType> {
    CLASSIC(0, "Classic checkpoint)"),
    MULTIPART(1, "Multipart checkpoint"),
    V2(2, "V2 checkpoint");

    private final int priority;
    private final String description;

    CheckpointType(int priority, String description) {
        this.priority = priority;
        this.description = description;
    }

    public int priority() { return priority; }

    @Override
    public int compareTo(CheckpointType o) {
        return Integer.compare(this.priority, o.priority);
    }
}

Reply:
Oh, it seems that one of the intention of this pr is kinda remove enum, so maybe have one at checkpoint level to simplify the checkpoint comparision? wdyt